benchmarks/tpcc/src/main/java/com/google/cloud/pgadapter/tpcc/HibernateBenchmarkRunner.java

```diff
@@ -0,0 +1,533 @@
+package com.google.cloud.pgadapter.tpcc;
+
+import com.google.cloud.pgadapter.tpcc.config.PGAdapterConfiguration;
+import com.google.cloud.pgadapter.tpcc.config.SpannerConfiguration;
+import com.google.cloud.pgadapter.tpcc.config.TpccConfiguration;
+import com.google.cloud.pgadapter.tpcc.entities.Customer;
+import com.google.cloud.pgadapter.tpcc.entities.CustomerId;
+import com.google.cloud.pgadapter.tpcc.entities.District;
+import com.google.cloud.pgadapter.tpcc.entities.DistrictId;
+import com.google.cloud.pgadapter.tpcc.entities.History;
+import com.google.cloud.pgadapter.tpcc.entities.HistoryId;
+import com.google.cloud.pgadapter.tpcc.entities.Item;
+import com.google.cloud.pgadapter.tpcc.entities.NewOrder;
+import com.google.cloud.pgadapter.tpcc.entities.Order;
+import com.google.cloud.pgadapter.tpcc.entities.OrderId;
+import com.google.cloud.pgadapter.tpcc.entities.OrderLine;
+import com.google.cloud.pgadapter.tpcc.entities.OrderLineId;
+import com.google.cloud.pgadapter.tpcc.entities.Stock;
+import com.google.cloud.pgadapter.tpcc.entities.StockId;
+import com.google.cloud.pgadapter.tpcc.entities.Warehouse;
+import com.google.cloud.spanner.Dialect;
+import jakarta.persistence.criteria.CriteriaBuilder;
+import jakarta.persistence.criteria.CriteriaQuery;
+import jakarta.persistence.criteria.JoinType;
+import jakarta.persistence.criteria.Root;
+import jakarta.persistence.criteria.Subquery;
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.sql.SQLException;
+import java.sql.Timestamp;
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Random;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+
+public class HibernateBenchmarkRunner extends AbstractBenchmarkRunner {
+
+  private final SessionFactory sessionFactory;
+  private final ThreadLocal<Session> sessionThreadLocal = new ThreadLocal<>();
```

Comment:
This makes the session live for the entire lifetime of the benchmark. That is too long. A session can best be compared with a (business) transaction. So typically one session is used to execute one database transaction, or at most a few, and should have a lifespan of at most a few seconds.

So it is probably better to refactor this so that each new transaction (e.g. each call to `newOrder()`, `payment()` etc.) creates a new session, and closes the session at the end of the method.